engine/engine.go

```diff
@@ -175,6 +175,10 @@ func New(opts Opts) *compatibilityEngine {
 		),
 	}
 
+	if opts.Reg != nil {
+		opts.EngineOpts.Reg = prometheus.WrapRegistererWithPrefix("compatibility_engine_", opts.Reg)
```

Comment:
I am not sure if `compatibility_engine_` is the because this is an implementation detail of how we manage to support all promql functions. Can we use `thanos_engine_` or whatever the prefix is for the main engine?

Reply:
So while working with https://github.com/thanos-io/thanos/pull/6234, we noticed that we would basically have 3 engines: the new engine, the new engine's compatibility prometheus engine, and the regular prometheus engine in Thanos. The metrics between the compatibility prometheus engine, and the prometheus engine run in Thanos clash when registering, so the `compatibility_engine` prefix is only added to the compatibility prometheus engine. And we reserve the `thanos_engine` prefix for metrics exported by new engine.

I wonder if there is some better way of handling this. Seems like we are running duplicative engines.

Reply:
Right, but what does `compatibility_engine` mean to someone who is using the library?

Reply:
Alternatively, we can have a constructor where a user injects a `QueryEngine` so we don't need to instantiate one internally.

Reply:
Hmm. For users of this library, I think knowing which queries hit fallback engine, might be somewhat desirable, but we already have a metric for that. 

I think the latter constructor approach might be preferable here.

Reply:
👍 let's try the new constructor then and see if it fixes the problem.

Reply:
Does that mean adding this prefix at `QueryEngineFactory` in thanos, not here?

Reply:
If I understood correctly, let's pass an instance of the original Prometheus PromQL engine to the Thanos PromQL engine via the constructor. We can re-use the same, old one, no need to create a new one with a prefix this way.